### PR TITLE
Adds license check in CI

### LINKED ISSUE #2077: Our CI should check that all library files include the proper licence header
As part of the cleanups in `devel` branch, I standardized the licence headers in all of the library (as opposed to test) files.

The new header is, as contributing.md says,
```cpp
    
    //              Copyright Catch2 Authors
    // Distributed under the Boost Software License, Version 1.0.
    //   (See accompanying file LICENSE_1_0.txt or copy at
    //        https://www.boost.org/LICENSE_1_0.txt)

    // SPDX-License-Identifier: BSL-1.0
```

We should check that all files use this licence header as part of automatic per-commit CI.

### PR DESCRIPTION
**Pull request by innerout**
## Description
Adds checkLicense.py that runs on a per-commit basis in CI checking the license for all source files.  

## GitHub Issues
Closes #2077

### CODE REVIEW

File: tools/scripts/checkLicense.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+def extract_license(split_license):
+    assert split_license != None
+
+    license = ""
+
+    for line in split_license:
+        license += line
+
+    return license
```
Comment (horenmar) [inline tools/scripts/checkLicense.py]: You can simplify this into single line, but at that point you might just get rid of the function altogether.

```suggestion
    return ''.join(split_license)
```

File: tools/scripts/checkLicense.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env python3
+import os
+import sys
+
+
+def get_license():
+    f = open("src/catch2/catch_all.hpp", "r")
+    license = "".join(f.readlines()[0:7])
+    f.close()
```
Comment (horenmar) [inline tools/scripts/checkLicense.py]: You should use context managers for handling closing files automatically.

File: tools/scripts/checkLicense.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+
+def check_license(license):
+    failed = 0
+    base_dir = "src/catch2/"
+
+    # The _ represents the list of directories in base_dir
+    for root, _, files in os.walk(base_dir):
+        for file in files:
+            f = open(root + "/" + file, "r")
```
Comment (horenmar) [inline tools/scripts/checkLicense.py]: Same as above.

File: tools/scripts/checkLicense.py (hunk 1 above)
Comment (innerout) [reply tools/scripts/checkLicense.py]: Fixed, thanks again for teaching me something new!

File: tools/scripts/checkLicense.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+            file_license = "".join(f.readlines()[0:7])
```
Comment (horenmar) [inline tools/scripts/checkLicense.py]: Thinking about this some more:
Why turn this into a string? Python lists are comparable, so you can just compare the slices directly.

Comment (innerout) [reply tools/scripts/checkLicense.py]: Well managing one big string is easier than managing many small ones for me. But you are right the code is probably better without the `join`.

File: .github/workflows/validate-header-guards.yml [hunk 5]
```diff
@@ -34,3 +34,6 @@ jobs:
       - name: checknames
         run: |
           python tools/scripts/checkDuplicateFilenames.py
+      - name: checklicense
```
Comment (horenmar) [inline .github/workflows/validate-header-guards.yml:37]: As a side note, this doesn't have to be so simple. The value for `name` key is a string, so you can write more descriptive name, e.g. "Check that files use proper license header", like so

```yaml
name: Check that files use proper license header
```

**Pull request merged**